Update Optiplex documentation

nedpfeiffer opened the pull request:
Walking through the recovery process I noticed some issues with the Optiplex documentation, namely:

- The Optiplex branch has been merged with Dasharo and coreboot upstream, so no need to mention it in `building-manual.md`
- `uefi-firmware-parser` only works with version `1.9` for whatever reason. I'll include logs of the errors I encountered below if it's helpful.
- There was a typo in the file path for the EC firmware which made it annoying to copy and paste

Here's that error I encountered with the latest version of `uefi-firmware` (`1.11`):

```
  Dell PFSSection: 7439ed9e-70d3-4b65-9e33-1963a7ad3c37 spec 0x01 ts 0x00 type 0x00 version .8.1.72.3002 size 0x77b148 (7844168 bytes)
    ME Container type= 0x10 version= 0x20 size= 0x30 (48 bytes) entires= 23 flags= 0xfffffc01
Traceback (most recent call last):
  File "/home/user/Downloads/venv/bin/uefi-firmware-parser", line 196, in <module>
    _process_show_extract(firmware)
  File "/home/user/Downloads/venv/bin/uefi-firmware-parser", line 19, in _process_show_extract
    parsed_object.showinfo('')
  File "/home/user/Downloads/venv/lib/python3.11/site-packages/uefi_firmware/__init__.py", line 130, in showinfo
    self.objs[i].showinfo(ts, i)
  File "/home/user/Downloads/venv/lib/python3.11/site-packages/uefi_firmware/pfs.py", line 422, in showinfo
    section.showinfo("%s  " % ts)
  File "/home/user/Downloads/venv/lib/python3.11/site-packages/uefi_firmware/pfs.py", line 324, in showinfo
    sub_object.showinfo("%s  " % ts)
  File "/home/user/Downloads/venv/lib/python3.11/site-packages/uefi_firmware/base.py", line 198, in showinfo
    self.object.showinfo(ts)
  File "/home/user/Downloads/venv/lib/python3.11/site-packages/uefi_firmware/me.py", line 675, in showinfo
    partition.showinfo("  %s" % ts)
  File "/home/user/Downloads/venv/lib/python3.11/site-packages/uefi_firmware/me.py", line 625, in showinfo
    purple(self.structure.Name.decode("utf-8")), purple(self.structure.Owner),
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xe0 in position 0: invalid continuation byte
```

Hope this is helpful, thanks!

nedpfeiffer commented:
Two things:

- An error is being thrown by pre-commit.ci due to line lengths being too long, but these long lines are necessary because the extracted firmware file paths are deeply nested.
- After running 'uefi-firmware-parser` I'm not able to find the BIOS ACM file listed in the documentation, although I can find the EC firmware just fine. Has the BIOS ACM file path changed since these docs were last updated? See below:

```
ned@fedora:~/Downloads/_O7010A29.exe.extracted/65C10_output/pfsobject/section-7ec6c2b0-3fe3-42a0-a316-22dd0517c1e8/volume-0x50000$ ls | grep file-d38
file-d386beb8-4b54-4e69-94f5-06091f67e0d3
ned@fedora:~/Downloads/_O7010A29.exe.extracted/65C10_output/pfsobject/section-7ec6c2b0-3fe3-42a0-a316-22dd0517c1e8/volume-0x50000$ ls | grep file-2d2
ned@fedora:~/Downloads/_O7010A29.exe.extracted/65C10_output/pfsobject/section-7ec6c2b0-3fe3-42a0-a316-22dd0517c1e8/volume-0x50000$ cd
ned@fedora:~$ # EC firmware is there, but what's the path for the BIOS ACM file?
```

Thanks!

pietrushnic commented:
@nedpfeiffer, thank you for your contribution. Would you mind fixing pre-commit issues?

nedpfeiffer commented:
@pietrushnic I've fixed the issue with the line lengths and trailing white space

miczyg1 commented on docs/variants/dell_optiplex/building-manual.md:
````diff
@@ -34,12 +34,6 @@ the result binary using `cbfstool`. The methods are covered later on in the
     cd coreboot
     ```
 
-    Replace `vX.Y.Z` with a valid version, eg. `v0.1.0`:
-
-    ```bash
-    git checkout optiplex_7010_9010_vX.Y.Z
-    ```
-
````
@nedpfeiffer it has to kept. It is not a branch, but a tag. We always encourage to build from tags for the stable and tested build in `building-manual.md` files.

nedpfeiffer replied:
Thanks for your patience, I restored the deleted documentation

filipleple reviewed (approved):
LGTM

nedpfeiffer commented:
To clarify, what is the correct path for the BIOS ACM file? The file path in the documentation doesn't exist:

```
ned@fedora:~/Downloads/_O7010A29.exe.extracted/65C10_output/pfsobject/section-7ec6c2b0-3fe3-42a0-a316-22dd0517c1e8/volume-0x50000$ ls | grep file-d38
file-d386beb8-4b54-4e69-94f5-06091f67e0d3
ned@fedora:~/Downloads/_O7010A29.exe.extracted/65C10_output/pfsobject/section-7ec6c2b0-3fe3-42a0-a316-22dd0517c1e8/volume-0x50000$ ls | grep file-2d2
ned@fedora:~/Downloads/_O7010A29.exe.extracted/65C10_output/pfsobject/section-7ec6c2b0-3fe3-42a0-a316-22dd0517c1e8/volume-0x50000$ cd
ned@fedora:~$ # EC firmware is there, but what's the path for the BIOS ACM file?
```

miczyg1 commented:
> To clarify, what is the correct path for the BIOS ACM file? The file path in the documentation doesn't exist:
> 
> ```
> ned@fedora:~/Downloads/_O7010A29.exe.extracted/65C10_output/pfsobject/section-7ec6c2b0-3fe3-42a0-a316-22dd0517c1e8/volume-0x50000$ ls | grep file-d38
> file-d386beb8-4b54-4e69-94f5-06091f67e0d3
> ned@fedora:~/Downloads/_O7010A29.exe.extracted/65C10_output/pfsobject/section-7ec6c2b0-3fe3-42a0-a316-22dd0517c1e8/volume-0x50000$ ls | grep file-2d2
> ned@fedora:~/Downloads/_O7010A29.exe.extracted/65C10_output/pfsobject/section-7ec6c2b0-3fe3-42a0-a316-22dd0517c1e8/volume-0x50000$ cd
> ned@fedora:~$ # EC firmware is there, but what's the path for the BIOS ACM file?
> ```

It is documented here: https://docs.dasharo.com/variants/dell_optiplex/initial-deployment/#obtaining-the-blobs

miczyg1 commented:
When I extract the EXE on my machine it gives me:

```
[miczyg@3M08 ~]$ ls _O7010A29.exe.extracted/65C10_output/pfsobject/section-7ec6c2b0-3fe3-42a0-a316-22dd0517c1e8/volume-0x500000/ |grep file-2d2
file-2d27c618-7dcd-41f5-bb10-21166be7e143
[miczyg@3M08 ~]$ ls _O7010A29.exe.extracted/65C10_output/pfsobject/section-7ec6c2b0-3fe3-42a0-a316-22dd0517c1e8/volume-0x500000/file-2d27c618-7dcd-41f5-bb10-21166be7e143/
file.obj  object-0.raw
```
